src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -836,6 +836,52 @@ void C2_MacroAssembler::fast_unlock_lightweight(Register obj, Register reg_rax,
   // C2 uses the value of ZF to determine the continuation.
 }
 
+#ifdef ASSERT
+void C2_MacroAssembler::checked_cast_int(const TypeInt* type, Register dst) {
+  BLOCK_COMMENT("CastII {");
+  Label fail;
+  Label succeed;
+  cmpl(dst, type->_lo);
```

Comment:
Optimization idea: some range checks may be redundant (when `lo`/`hi` hold min/max values).